universal-login-wallet/src/core/presenters/WalletPresenter.ts

```diff
@@ -5,6 +5,16 @@ export class WalletPresenter {
     private walletService: WalletService
   ) {}
 
+  getPrivateKey() {
+    switch (this.walletService.state.kind) {
+      case 'Connecting':
+      case 'Deployed':
+        return this.walletService.state.wallet.privateKey;
+      default:
+        throw new Error('Invalid wallet state: expected Deployed or Connecting wallet');
+    }
+  }
+
```

Comment:
Not sure about it. Don't know the difference between  'Connecting' and 'Deployed'